### PR TITLE
design-proposal: Network binding plugin

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

The network-binding-plugin feature design started back in mid 2023 and continued to evolve until implemented.

The design was conducted on a google document [1] out of convenience. (access is limited to the `kubevirt-dev` google group)

We now migrate it to a markdown version to provide better visibility.

[1] https://docs.google.com/document/d/13fLaqZuM8El_CT5XzxPv-1epKsB1-Spv_jtXveEkkdQ

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This is converting a google document to markdown, all design content has been already reviewed thoroughly
and implemented.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: design-proposals/network-binding-plugin/network-binding-plugin.md [hunk 1]
```diff
@@ -0,0 +1,943 @@
...
+        - All other bindings will get deprecated in favor of the plugin
+          alternative.
+
+### Feature Gate
+
+The network binding plugin API and integration points should be
+initially protected by a FG.
+Feature lifecycle is expected to pass alpha, beta and eventually to GA.
+
+## Appendix B: Concrete example for bridge-binding
```
[the remark this answers is not in the document]
Comment (EdDev) [reply design-proposals/network-binding-plugin/network-binding-plugin.md:345]: Done

**Review by ormergi — APPROVED** [review]

**Comment by ormergi** [comment]
Thanks!

**Pull request merged**